presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueriesGlueMetastore.java

```diff
@@ -0,0 +1,446 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.hive.HiveExternalWorkerQueryRunner;
+import com.facebook.presto.hive.HivePlugin;
+import com.facebook.presto.testing.ExpectedQueryRunner;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.MaterializedRow;
+import com.facebook.presto.testing.QueryRunner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.util.Locale.ENGLISH;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * TestHiveGlueMetastoreUsingThrift requires AWS credentials to be correctly set up in order to access Glue catalog.
+ * You can configure credentials locally in ~/.aws/credentials before running the test. Alternatively, you can set up
+ * AWS access/secret keys or IAM role in `setUpGlueCatalog` method in the test suite using the Glue config properties
+ * from https://prestodb.io/docs/current/connector/hive.html#aws-glue-catalog-configuration-properties.
+ */
+abstract class TestHiveQueriesGlueMetastore
+        extends TestHiveQueries
+{
+    private static final String TEMPORARY_PREFIX = "tmp_presto_glue_test_";
+    // Catalog name used in the tests, it is also hardcoded in HiveExternalWorkerQueryRunner so workers can register
+    // the connector so if you want to change the name, also update it in HiveExternalWorkerQueryRunner.
+    private static final String CATALOG_NAME = "test_glue";
+
+    // Warehouse directory for the Glue catalog.
+    private File tempWarehouseDir;
+    // Fully-qualified schema name in format [catalog].[schema] used in tests.
+    private String schemaName;
+
+    private final boolean useThrift;
```

Comment:
Why add another useThrift? It's already defined in the super class AbstractTestHiveQueries.

Reply:
Updated.